slackapi/deno-slack-hooks
# Migrate Deno Bundle to ESbuild

###  Summary

This PR aims to improve the behavior of the build hooks by adding a fallback to [esbuild](https://esbuild.github.io/api/#hot-reloading-js) if `deno bundle` fails.

#### Testing

1. Create a new sample application with the CLI
2. Pull this branch [migrate-deno-run-bundle](https://github.com/slackapi/deno-slack-hooks/tree/migrate-deno-run-bundle)
3. add the following to the `slack.json` file 
```json
"build": "deno run -q --config=deno.jsonc --allow-read --allow-write --allow-net --allow-run --allow-env path/to/local/src/build.ts",
```
4. Run `slack deploy -vv`
5. Everything should work as expected with `deno bundle`
6. In `src/build.ts` above line 81 add `throw new BundleError()` (this will force a fallback to esbuild)
7.  Run `slack deploy` in your app folder
8. You should see this warning when executing `Failed bundling with 'Deno Bundle' falling back to esbuild`
9. Bundling should work

#### Feedback

- General code cleanliness, it was tricky to figure out when errors should be raised
- User experience, should we warn the user that their app falls back to esbuild?
- If the esbuild bundle fails should we mention to the user that first the deno bundle failed and then the esbuild failed or mention bundling failed?

### Requirements
